superset/connectors/druid/models.py

```diff
@@ -941,23 +941,24 @@ def metrics_and_post_aggs(
         adhoc_agg_configs = []
         postagg_names = []
         for metric in metrics:
-            if utils.is_adhoc_metric(metric):
+            if isinstance(metric, dict) and utils.is_adhoc_metric(metric):
```

Comment:
I'm surprised this was necessary as it seems `is_adhoc_metric(metric)` can handle both a `dict` and `str`. I wonder in adding typing there, i.e., 

```python
is_adhoc_metric(metric: Union[Dict[str, Any], str]) -> bool:
```

would remove the need for `isinstance(metric, dict)` which is unnecessary.

Reply:
This and L946 are for function calls later in the current context. `Union[Dict, str]` was passed on to `adhoc_agg_configs `, `saved_agg_names` and `saved_agg_names`, but could not be accepted by `get_aggregations()` and `resolve_postagg()`, which only accept either just `dict` or just `str`.

Apparently `mypy` was not smart enough to figure our `metric` type just by the outcome of `is_adhoc_metric()`.